### PR TITLE
Implement tests for *presigned_url* methods.

### PR DESCRIPTION
**Pull request by golenkovm**
Issue #176

### CODE REVIEW

File: tests/tool_objectfs_testcase.php [hunk 1]
```diff
@@ -252,4 +252,13 @@ protected function is_externally_readable_by_url($url) {
             return false;
         }
     }
+
+    protected function integration_client_supports_presigned_urls($client) {
+        $supportedlist = array();
+        $supportedlist[] = 'tool_objectfs\tests\test_s3_integration_client';
```
Comment (dmitriim) [inline tests/tool_objectfs_testcase.php]: Can you please explain why you check that the client supports presigned urls amnually and not using support_presigned_urls method?

File: tests/classes/test_client.php [hunk 2]
```diff
@@ -108,6 +108,7 @@ public function get_maximum_upload_size() {
     public function support_signed_urls() {
         return false;
     }
+
     public function generate_signed_url($contenthash) {
         return 'Not supported';
```
Comment (brendanheywood) [inline tests/classes/test_client.php]: Not quite related to your patch but we should change this to throw a 'not implemented' exception ?

Comment (golenkovm) [reply tests/classes/test_client.php]: Oh, thank you for pointing on that. We use `generate_presigned_url()` method and it throws exception. `generate_signed_url()` was the first version of this method and not used anymore. I will remove it in further commit.

File: tests/tool_objectfs_testcase.php (hunk 1 above)
Comment (golenkovm) [reply tests/tool_objectfs_testcase.php]: You're right, it's better to use `$externalclient->support_presigned_urls()` method. Re-written tests will be in further commit.

File: tests/object_file_system_test.php [hunk 3]
```diff
@@ -563,35 +562,41 @@ public function test_presigned_url_should_redirect_method_return_true_regardless
...
-            $object = $this->create_local_object();
-            $this->assertFalse($this->filesystem->presigned_url_should_redirect($object->contenthash));
+        if (!$externalclient->support_presigned_urls()) {
+            $this->markTestSkipped('Pre-signed URLs not supported for given storage.');
         }
+
+        $this->set_externalclient_config('enablepresignedurls', '1');
+        $this->set_externalclient_config('presignedminfilesize', 1000);
+        $object = $this->create_local_object();
+        $this->assertFalse($this->filesystem->presigned_url_should_redirect($object->contenthash));
```
Comment (dmitriim) [inline tests/object_file_system_test.php]: why is it false here? does create_local_object create the file less than 1000? If it is, can you please add some comment here?

File: tests/object_file_system_test.php [hunk 4]
```diff
@@ -563,35 +562,41 @@ public function test_presigned_url_should_redirect_method_return_true_regardless
...
-            $this->set_externalclient_config('enablepresignedurls', '1');
-            $this->set_externalclient_config('presignedminfilesize', 0);
-            $object = $this->create_local_object();
-            $this->assertTrue($this->filesystem->presigned_url_should_redirect($object->contenthash));
+        if (!$externalclient->support_presigned_urls()) {
+            $this->markTestSkipped('Pre-signed URLs not supported for given storage.');
         }
+
+        $this->set_externalclient_config('enablepresignedurls', '1');
+        $this->set_externalclient_config('presignedminfilesize', 0);
```
Comment (dmitriim) [inline tests/object_file_system_test.php]: Can you please also test scenarios when presignedminfilesize and enablepresignedurls are set to null, empty string, '0' as a string?

File: tests/object_file_system_test.php (hunk 3 above)
Comment (golenkovm) [reply tests/object_file_system_test.php]: It is false because local test files are created with file size = 10:
https://github.com/catalyst/moodle-tool_objectfs/blob/issue176/tests/tool_objectfs_testcase.php#L86
